services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -5432,10 +5436,31 @@ private Pair<ExecutionStatus, String> getDaVinciPushStatusAndDetails(
         incompletePartition.add(partitionId);
       }
     }
+
+    if (totalInstanceCount == 0) {
+      return new Pair<>(completeStatus, "No DaVinci instance found yet.");
+    }
+
+    // Report error if too many davinci instances are not alive for over 5 mins
+    if (liveInstanceCount < 0.5 * totalInstanceCount) {
```

Comment:
Potentially we need to specify some minimal threshold, such as the totalInstanceCount needs to be large than some number (such as 3 or 5), otherwise, a rolling deployment may trigger a false positive event.